src/pages/blog/http-and-lambda.mdx

```diff
@@ -0,0 +1,338 @@
+| Invalid JSON response | `200`         | `{`                       | The JSON formatted request |
+| Exception thrown      | `502`         | `Internal Server Error`   | The JSON formatted request |
+| POST Method disabled  | `403`         | HTML explaining the error | Nothing                    |
+
+
+
+## ALB
+
+Creating an **Internet-facing** ALB with a Target Group pointing at a Lambda
+was [simple enough](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/create-application-load-balancer.html).
+As ever when configuring ALBs I ran into an issue where my Security Group was
+misconfigured and my tests were timing out, but once that was resolved
+everything worked just fine.
+
+The [request](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/lambda-functions.html#receive-event-from-load-balancer)
+and [response](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/lambda-functions.html#respond-to-load-balancer)
+formats are similar but slightly different to the Function URL formats. One
+major difference is that ALBs are more *strict* with the response format.
+
+Because of this strictness I wanted to test the case where the response
+**format** was valid too, so I modified the lambda code like so:
+
+```clojure
+(ns lambda
+  (:gen-class
+   :implements [com.amazonaws.services.lambda.runtime.RequestStreamHandler]))
+
+(defn -handleRequest [_ is os context]
+  (let [input (slurp is)]
+    (println "input:" input)
+    (when (re-find #"error" input)
+      (throw (ex-info "test" {:my "error"})))
+    (cond
+      (re-find #"invalid" input) (spit os "{")
+      ;; 👇 Added an option to return JSON in the format that the ALB wants
+      (re-find #"valid" input) (spit os "{\"isBase64Encoded\":false,\"statusCode\":200}")
+      :else (spit os "{}"))))
+```
+
+And an additional test using the command: `curl -v -XPOST --data '{}' --header 'Test: valid' <url under test>`
+
+| Test                  | Status Code   | Response Body             | `input` in lambda          |
+| ---                   | ---           | ---                       | ---                        |
+| Valid JSON request    | `502`         | HTML for 502 Bad Gateway  | The JSON formatted request |
+| Invalid JSON request  | `502`         | HTML for 502 Bad Gateway  | The JSON formatted request |
+| Invalid JSON response | `502`         | HTML for 502 Bad Gateway  | The JSON formatted request |
+| Exception thrown      | `502`         | HTML for 502 Bad Gateway  | The JSON formatted request |
+| Valid response format | `200`         | (an empty body)           | The JSON formatted request |
+
+As mentioned the response format is very strict, failing to use it means an
+obscure `502 Bad Gateway` error. So make sure to catch exceptions and always
+return the correct JSON.
+
+
+
+## API Gateway w/ HTTP API
+
+HTTP APIs are API Gateway's simplest option, which is reflected in it's setup:
```

Comment:
Consider re-phrasing to intro API Gateway a little more (since it's not been mentioned for quite a while by this point), e.g. 'The API Gateway method presents a few options, but using HTTP APIs is the simplest option`